server/api/handler/transaction.go

```diff
@@ -0,0 +1,301 @@
+	"github.com/teamhanko/passkey-server/persistence/models"
+	"github.com/teamhanko/passkey-server/persistence/persisters"
+	"net/http"
+	"time"
+)
+
+type transactionHandler struct {
+	*webauthnHandler
+}
+
+func NewTransactionHandler(persister persistence.Persister) WebauthnHandler {
+	webauthnHandler := newWebAuthnHandler(persister)
+
+	return &transactionHandler{webauthnHandler}
+}
+
+func (t *transactionHandler) Init(ctx echo.Context) error {
+	dto, err := BindAndValidateRequest[request.InitTransactionDto](ctx)
+	if err != nil {
+		return err
+	}
+
+	h, err := helper.GetHandlerContext(ctx)
+	if err != nil {
+		ctx.Logger().Error(err)
+		return err
+	}
+
+	transactionModel, err := dto.ToModel()
+	if err != nil {
+		ctx.Logger().Error(err)
+		return echo.NewHTTPError(http.StatusInternalServerError, "unable to process request")
+	}
+
+	return t.persister.GetConnection().Transaction(func(tx *pop.Connection) error {
+		webauthnUser, err := t.persister.GetWebauthnUserPersister(tx).GetByUserId(dto.UserId, h.Tenant.ID)
+		if err != nil {
+			ctx.Logger().Error(err)
+			return echo.NewHTTPError(http.StatusNotFound, "Unable to find user")
+		}
+
+		if webauthnUser == nil {
+			return echo.NewHTTPError(http.StatusNotFound, "Unable to find user")
+		}
+
+		assertion, sessionData, err := h.Webauthn.BeginLogin(
+			intern.NewWebauthnUser(*webauthnUser),
+			webauthn.WithUserVerification(h.Config.WebauthnConfig.UserVerification),
+			t.withTransaction(transactionModel.Identifier, transactionModel.Data),
+		)
+		if err != nil {
+			auditErr := h.AuditLog.CreateWithConnection(tx, models.AuditLogWebAuthnTransactionInitFailed, &webauthnUser.UserID, transactionModel, err)
+			if auditErr != nil {
+				ctx.Logger().Error(auditErr)
+				return fmt.Errorf(auditlog.CreationFailureFormat, auditErr)
+			}
+
+			ctx.Logger().Error(err)
+			return echo.NewHTTPError(
+				http.StatusBadRequest,
+				fmt.Errorf("failed to create webauthn assertion options for transaction: %w", err),
+			)
+		}
+
+		// workaround: go-webauthn changes only the assertion challenge when giving LoginOptions
+		sessionData.Challenge = assertion.Response.Challenge.String()
+		transactionModel.Challenge = sessionData.Challenge
+		transactionModel.WebauthnUserID = webauthnUser.ID
+		transactionModel.TenantID = h.Tenant.ID
+
+		err = t.persister.GetTransactionPersister(tx).Create(transactionModel)
+		if err != nil {
+			auditErr := h.AuditLog.CreateWithConnection(tx, models.AuditLogWebAuthnAuthenticationInitFailed, &webauthnUser.UserID, transactionModel, err)
+			if auditErr != nil {
+				ctx.Logger().Error(auditErr)
+				return fmt.Errorf(auditlog.CreationFailureFormat, auditErr)
+			}
+
+			ctx.Logger().Error(err)
+			return fmt.Errorf("failed to store webauthn transaction data: %w", err)
+		}
+
+		err = t.persister.GetWebauthnSessionDataPersister(tx).Create(*intern.WebauthnSessionDataToModel(sessionData, h.Tenant.ID, models.WebauthnOperationAuthentication))
```

Comment:
We should add a new Operation name `transaction` instead of `authentication` for the transaction.

Reply:
Fixed